README.md

```diff
@@ -0,0 +1,34 @@
+<div align="center">
+  <img src="./brand/dragonfly-rainbow.svg" width="300">
+</div>
+
+# Hypothesis
+
+* [Website](https://hypothesis.works/)
+* [Documentation](https://hypothesis.readthedocs.io/en/latest/)
+* [Source code](https://github.com/hypothesisWorks/hypothesis/)
+* [Contributing](https://github.com/HypothesisWorks/hypothesis/blob/master/CONTRIBUTING.rst)
+* [Community](https://hypothesis.readthedocs.io/en/latest/community.html)
+
+Hypothesis is the canonical property-based testing library for Python. A property-based test asserts something for *all* inputs, and lets Hypothesis generate them — including inputs you may not have thought of.
+
+```python
+from hypothesis import given, strategies as st
+
+
+@given(st.lists(st.integers() | st.floats()))
+def test_matches_builtin(lst):
+    assert sorted(lst) == my_sort(lst)
+```
+
+Additionally, when a property fails, Hypothesis doesn't just report any failing example — it reports the simplest possible one. This makes property-based tests a powerful tool for debugging, as well as testing.
+
+```python
+Falsifying example: test_matches_builtin(lst=[math.nan])
+```
```

Comment:
There's an unsaid "your sort function might break on nans" here which we might need to make explicit. I couldn't think of an example which was (compelling & short & contained a bug for shrinking). I'm not a fan of the math.mean example in the current readme; too much text to parse in the falsifying example, and `max_size=1` + `allow_nan=False` in the generator is a bit overwhelming for the reader the first time imo.

Reply:
FWIW... there's another possible confusion for the reader here, being that this particular assert failure won't actually happen:
```
sorted([math.nan]) == [math.nan]
```
whereas
```
sorted(float("nan")) != [float("nan")]
```
because list equality doesn't check logical equality of id-equal objects.

So, in practice, the sort function wouldn't break before there's two elements in the list, e.g.,
```
my_sort = lambda l: list(reversed(sorted(l, key=lambda a: -a)))
```
→
```
Falsifying example: test(
    l=[0, -nan],
)
```
(don't know why `-nan` rather than `nan`)

Reply:
I created #4277 about that `-nan` btw

Reply:
yeah I had [0, math.nan] originally, probably we should go back to that 👍. I'll see if working in that concrete `my_sort` definition flows well too.

Reply:
I don't necessarily mean the concrete definition is necessary, just that the leading zero is necessary to make the implied bug plausible.